extensions/splunk/tests/MockSplunkHEC.h

```diff
@@ -0,0 +1,214 @@
+ protected:
+  bool handlePostImpl(struct mg_connection* conn) override {
+    constexpr const char * body = "{\"text\":\"Success\",\"code\":0,\"ackId\":808}";
+    mg_printf(conn, "HTTP/1.1 200 OK\r\n");
+    mg_printf(conn, "Content-length: %lu", strlen(body));
+    mg_printf(conn, "\r\n\r\n");
+    mg_printf(conn, body);
+    return true;
+  }
+};
+
+class AckIndexerHandler : public MockSplunkHandler {
+ public:
+  explicit AckIndexerHandler(std::string token, std::vector<uint64_t> indexed_events, std::function<void(const struct mg_request_info *request_info)>& assertions)
+      : MockSplunkHandler(std::move(token), assertions), indexed_events_(indexed_events) {}
+
+ protected:
+  bool handlePostImpl(struct mg_connection* conn) override {
+    std::vector<char> data;
+    data.reserve(2048);
+    mg_read(conn, data.data(), 2048);
+    rapidjson::Document post_data;
+
+    rapidjson::ParseResult parse_result = post_data.Parse<rapidjson::kParseStopWhenDoneFlag>(data.data());
+    if (parse_result.IsError())
+      return sendInvalidFormat(conn);
+    if (!post_data.HasMember("acks") || !post_data["acks"].IsArray())
+      return sendInvalidFormat(conn);
+    std::vector<uint64_t> ids;
+    for (auto& id : post_data["acks"].GetArray()) {
+      ids.push_back(id.GetUint64());
+    }
+    rapidjson::Document reply = rapidjson::Document(rapidjson::kObjectType);
+    reply.AddMember("acks", rapidjson::kObjectType, reply.GetAllocator());
+    for (auto& id : ids) {
+      rapidjson::Value key(std::to_string(id).c_str(), reply.GetAllocator());
+      reply["acks"].AddMember(key, std::find(indexed_events_.begin(), indexed_events_.end(), id) != indexed_events_.end() ? true : false, reply.GetAllocator());
+    }
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    reply.Accept(writer);
+
+    mg_printf(conn, "HTTP/1.1 200 OK\r\n");
+    mg_printf(conn, "Content-length: %lu", buffer.GetSize());
+    mg_printf(conn, "\r\n\r\n");
+    mg_printf(conn, "%s" , buffer.GetString());
+    return true;
+  }
+
+  bool sendInvalidFormat(struct mg_connection* conn) {
+    constexpr const char * body = "{\"text\":\"Invalid data format\",\"code\":6}";
+    mg_printf(conn, "HTTP/1.1 400 Bad Request\r\n");
+    mg_printf(conn, "Content-length: %lu", strlen(body));
+    mg_printf(conn, "\r\n\r\n");
+    mg_printf(conn, body);
+    return true;
+  }
+
+  std::vector<uint64_t> indexed_events_;
+};
+
+class MockSplunkHEC {
+ public:
+  static constexpr const char* TOKEN = "Splunk 822f7d13-2b70-4f8c-848b-86edfc251222";
+
+  static inline std::vector<uint64_t> indexed_events = {0, 1};
+
+  explicit MockSplunkHEC(std::string port) : port_(std::move(port)) {
+    std::vector<std::string> options;
+    options.emplace_back("listening_ports");
+    options.emplace_back(port_);
+    server_.reset(new CivetServer(options, &callbacks_, &logger_));
```

Comment:
I think we need to initialize the civet library to safely use it like in TestServer.h

Reply:
good idea, fixed in https://github.com/apache/nifi-minifi-cpp/pull/1219/commits/412fa041455d782eedc52bdfb24c680ae7e56808#diff-22bed17b8af51ae7e6dba78a65e7a6c2a0e66fe55431bf0d27ee8508055afaa6R176-R193

Reply:
moving the `CivetLibrary` into its own header under `libminifi/test` might have its merits but I do not insist on it